File: GeneratedEntityReaderClasses/Classes/ConfigDamageCritical.cs
namespace GeneratedNoitaClasses;
public class ConfigDamageCritical :  iNoitaType<ConfigDamageCritical>{
	public NoitaInt chance { get; set; } = new();
	public NoitaFloat damage_multiplier { get; set; } = new();

    public void Read(NoitaStream s){
		chance.Read(s);
		damage_multiplier.Read(s);

    }
    public void Write(NoitaStream s){
		chance.Write(s);
		damage_multiplier.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/ConfigDamageCritical.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class ConfigDamageCritical:
    chance:NoitaInt = field(default_factory=NoitaInt)
    damage_multiplier:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.chance.Read(s)
        self.damage_multiplier.Read(s)
    
    def Write(self, s:NoitaStream):
        self.chance.Write(s)
        self.damage_multiplier.Write(s)

